typing/typecore.ml

```diff
@@ -7742,6 +7745,18 @@ and type_application env app_loc expected_mode position_and_mode
             collect_apply_args env funct ignore_labels ty (instance ty)
               (value_to_alloc_r2l funct_mode) sargs ret_tvar
           in
+          let sort_fully_applied =
+            (* [ty_ret] from [collect_apply_args] is the final return type of
+               the function after it has been applied to all arguments, even
+               those omitted in this application. This may be different from the
+               type of this application, if there are omitted arguments. We need
+               to know the sort of this final return type to compile such
+               applications. *)
+            match type_sort ~why:Function_result ~fixed:false env ty_ret with
```

Comment:
This call to `type_sort` makes me a little unhappy. Is there a way to get the information from `collect_apply_args`?